pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package persistentvolume
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	vol "k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+// PersistentVolumeController is a controller that synchronizes
+// PersistentVolumeClaims and PersistentVolumes. It starts two
+// framework.Controllers that watch PerstentVolume and PersistentVolumeClaim
+// changes.
+type PersistentVolumeController struct {
+	volumes          persistentVolumeOrderedIndex
+	volumeController *framework.Controller
+	claims           cache.Store
+	claimController  *framework.Controller
+	kubeClient       clientset.Interface
+
+	// Map of channels used to stop both PerstentVolume and
+	// PersistentVolumeClaim framework.Controllers (running in separate
+	// goroutines).
+	stopChannels map[string]chan struct{}
```

Comment:
minor, optional: Making this a map while the cache and controller fields are hand-unrolled is inconsistent.  I would just put the channels above next to the volumes* and claims*